src/app/api/services/edit-page/edit-page.service.ts

```diff
@@ -3,22 +3,151 @@ import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
 import { RequestMethod } from '@angular/http';
 import { DotRenderedPage } from '../../../portlets/dot-edit-page/shared/models/dot-rendered-page.model';
+import { PageMode } from '../../../portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component';
+import { DotEditPageState } from '../../../shared/models/dot-edit-page-state/dot-edit-page-state.model';
+import { DotRenderedPageState } from '../../../portlets/dot-edit-page/shared/models/dot-rendered-page-state.model';
 
 /**
  * Provide util methods to get a edit page html
  * @export
  * @class EditPageService
  */
+
 @Injectable()
 export class EditPageService {
     constructor(private coreWebService: CoreWebService) {}
 
-    get(url: string): Observable<DotRenderedPage> {
+    /**
+     * Get the page HTML in edit mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getEdit(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'EDIT_MODE');
+    }
+
+    /**
+     * Get the page HTML in preview mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getPreview(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'PREVIEW_MODE');
+    }
+
+    /**
+     * Get the page HTML in live mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getLive(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'LIVE_MODE');
+    }
+
+    /**
+     * Lock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    lock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/lock/inode/${inode}`
+        });
```

Comment:
I don't if we should return the ResponseView here, the end point return the follow:
{
      inode: inode,
      id:  id,
      message: "locked",
      return: 200
}

Reply:
It returns ```bodyJsonObject``` so I'm going to pick from that.

Reply:
it is a legacy resource, it would good to start doing a v1 @wezell :)